lib/serializer/StdInc.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * StdInc.h, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+#pragma once
+
+#include <typeinfo>
+#include <string>
+#include "CTypeList.h"
+
+template<class T, class F>
+inline const T * dynamic_ptr_cast(const F * ptr)
+{
+	#ifndef __APPLE__
+  return dynamic_cast<const T*>(ptr);

```

Comment:
please format whole file according to guidelines, https://wiki.vcmi.eu/Coding_guidelines